bootstrapper/internal/initserver/initserver.go

```diff
@@ -113,6 +115,9 @@ func (s *Server) Serve(ip, port string, cleaner cleaner) error {
 
 // Init initializes the cluster.
 func (s *Server) Init(ctx context.Context, req *initproto.InitRequest) (*initproto.InitResponse, error) {
+	s.shutdownLock.Lock()
```

Comment:
I think we should maintain the parallel nature of the server. Either by using a semaphore or a RWLock. 

The idea is to allow sending init requests to test the behavior for the checks before the nodeLock. Also this would emphasize, that the security is still based on the nodeLock and not the shutdownLock.